be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/Services/Products/ProductsService.cs

```diff
@@ -114,9 +114,27 @@ public async Task<PagedResults<IEnumerable<CatalogItemViewModel>>> GetProductsAs
             };
         }
 
-        public async Task<IEnumerable<string>> GetProductSuggestionsAsync(string searchTerm, int size, string language, string token)
+        public async Task<IEnumerable<ProductSuggestionViewModel>> GetProductSuggestionsAsync(string searchTerm, int size, string language, string token, string searchArea)
```

Comment:
**questionL** Wouldn't it be possible to do a normal select without waiting for the remaining tasks?